dir.rb

```diff
@@ -108,114 +194,157 @@ def self.open(name, encoding: nil, &block)
+  #
+  #   Dir.new('.') # => #<Dir:.>
+  #
+  # The value given with optional keyword argument +encoding+
+  # specifies the encoding for the directory entry names;
+  # if +nil+ (the default), the file system's encoding is used:
   #
-  # Returns a new directory object for the named directory.
+  #   Dir.new('.').read.encoding                       # => #<Encoding:UTF-8>
+  #   Dir.new('.', encoding: 'US-ASCII').read.encoding # => #<Encoding:US-ASCII>
   #
-  # The optional <i>encoding</i> keyword argument specifies the encoding of the directory.
-  # If not specified, the filesystem encoding is used.
   def initialize(name, encoding: nil)
     Primitive.dir_initialize(name, encoding)
   end
 
   # call-seq:
-  #    Dir[ string [, string ...] [, base: path] [, sort: true] ] -> array
+  #   Dir[*patterns, base: nil, sort: true] -> array
+  #
+  # Calls Dir.glob with argument +patterns+
+  # and the values of keyword arguments +base+ and +sort+;
+  # returns the array of selected entry names.
   #
-  # Equivalent to calling
-  # <code>Dir.glob([</code><i>string,...</i><code>], 0)</code>.
   def self.[](*args, base: nil, sort: true)
     Primitive.dir_s_aref(args, base, sort)
   end
 
   # call-seq:
-  #    Dir.glob( pattern, [flags], [base: path] [, sort: true] )                       -> array
-  #    Dir.glob( pattern, [flags], [base: path] [, sort: true] ) { |filename| block }  -> nil
+  #   Dir.glob(*patterns, _flags = 0, flags: _flags, base: nil, sort: true) -> array
+  #   Dir.glob(*patterns, _flags = 0, flags: _flags, base: nil, sort: true) {|entry_name| ... } -> nil
   #
-  # Expands +pattern+, which is a pattern string or an Array of pattern
-  # strings, and returns an array containing the matching filenames.
-  # If a block is given, calls the block once for each matching filename,
-  # passing the filename as a parameter to the block.
+  # Forms an array _entry_names_ of the entry names selected by the arguments.
   #
-  # The optional +base+ keyword argument specifies the base directory for
-  # interpreting relative pathnames instead of the current working directory.
-  # As the results are not prefixed with the base directory name in this
-  # case, you will need to prepend the base directory name if you want real
-  # paths.
+  # Argument +patterns+ is a pattern string or an array of pattern strings;
+  # note that these are not regexps.
   #
-  # The results which matched single wildcard or character set are sorted in
-  # binary ascending order, unless +false+ is given as the optional +sort+
-  # keyword argument.  The order of an Array of pattern strings and braces
-  # are preserved.
+  # With no block, returns array _entry_names_;
+  # example (using the {simple file tree}[rdoc-ref:Dir@About+the+Examples]):
   #
-  # Note that the pattern is not a regexp, it's closer to a shell glob.
-  # See File::fnmatch for the meaning of the +flags+ parameter.
-  # Case sensitivity depends on your system (+File::FNM_CASEFOLD+ is ignored).
```

Comment:
We should talk about the possible `flags` found in `File::Constants`.

Reply:
I had not known where FNM_* were.

Reply:
I think we should list the flags under `File::Constants` that can be used in this method.

Reply:
Ok I guess it's fair that this isn't causing a regression

Reply:
@peterzhu2118, I will not forget about your suggestion.

Reply:
@peterzhu2118, which flags in File::Constants apply for Dir.glob?  I tried a couple of unlikelies (TRUNC, CREAT) and got no error.

Reply:
It looks like invalid flags just get silently ignored.